### PR TITLE
testbuild.sh: Add an option to show the list of configs

### PR DESCRIPTION
**Pull request by yamt**
This was useful for me when tweaking testlists.

### CODE REVIEW

File: tools/testbuild.sh [hunk 1]
```diff
@@ -196,12 +201,21 @@ function build {
 # Coordinate the steps for the next build test
 
 function dotest {
-  echo "===================================================================================="
+  if [ ${PRINTLISTONLY} -ne 1 ]; then
```
Comment (xiaoxiang781216) [inline tools/testbuild.sh:204]: Should we add a function to avoid the code dup here and line 281?

File: tools/testbuild.sh [hunk 2]
```diff
@@ -196,12 +201,21 @@ function build {
...
+    echo "===================================================================================="
+  fi
   config=`echo $1 | cut -d',' -f1`
   re="-${config/\//:}[[:space:]]"
   if [[ "${blacklist} " =~ $re ]]; then
+    if [ ${PRINTLISTONLY} -eq 1 ]; then
```
Comment (xiaoxiang781216) [inline tools/testbuild.sh:210]: Why  don't print the skip entry? it's also important to monitor and reduce them.

File: tools/testbuild.sh [hunk 3]
```diff
@@ -196,12 +201,21 @@ function build {
...
+      return
+    fi
     echo "Skipping: $1"
   else
+    if [ ${PRINTLISTONLY} -eq 1 ]; then
```
Comment (xiaoxiang781216) [inline tools/testbuild.sh:215]: Move after line 220?

Comment (yamt) [reply tools/testbuild.sh:215]: why?

File: tools/testbuild.sh (hunk 2 above)
Comment (yamt) [reply tools/testbuild.sh:210]: because it was useless for my purpose.
if you have a use case, i can add it back.

File: tools/testbuild.sh (hunk 1 above)
Comment (yamt) [reply tools/testbuild.sh:204]: do you mean something like this?

    function print_separator {
      if [ ${PRINTLISTONLY} -ne 1 ]; then
          echo "===================================================================================="
      fi
    }

File: tools/testbuild.sh (hunk 3 above)
Comment (xiaoxiang781216) [reply tools/testbuild.sh:215]: Line 219 already print the config name, it doesn't make sense to dup the code here.

File: tools/testbuild.sh (hunk 2 above)
Comment (xiaoxiang781216) [reply tools/testbuild.sh:210]: We need monitor configs we skip carefully and reduce them as much as possible, so this log is useful.

File: tools/testbuild.sh (hunk 1 above)
Comment (xiaoxiang781216) [reply tools/testbuild.sh:204]: Yes.

File: tools/testbuild.sh [hunk 4]
```diff
@@ -92,6 +94,9 @@ while [ ! -z "$1" ]; do
     shift
     nuttx="$1"
     ;;
+  -p )
+    PRINTLISTONLY=1
```
Comment (xiaoxiang781216) [inline tools/testbuild.sh:98]: Should we use lower case?

Comment (yamt) [reply tools/testbuild.sh:98]: why? the majority of variables in this script are with upper case.

File: tools/testbuild.sh (hunk 3 above)
Comment (yamt) [reply tools/testbuild.sh:215]: i wanted machine-readable output. maybe it's a matter of preference.

Comment (yamt) [reply tools/testbuild.sh:211]: i reverted the output changes to keep the change minimum. i guess you prefer this version.

**Pull request merged**